bazel/envoy_build_system.bzl

```diff
@@ -20,12 +20,45 @@ def envoy_copts(repository, test = False):
         repository + "//bazel:dbg_build": ["-ggdb3"],
     }) + select({
         repository + "//bazel:disable_tcmalloc": [],
+        "//bazel/osx:darwin": [],
         "//conditions:default": ["-DTCMALLOC"],
     }) + select({
         repository + "//bazel:disable_signal_trace": [],
         "//conditions:default": ["-DENVOY_HANDLE_SIGNALS"],
     })
 
+# Compute the final linkopts based on various options.
+def envoy_linkopts():
+    return select({
+        "//bazel/osx:darwin": [],
```

Comment:
I think one could inject a custom section into the mach-o executable by adding a flag to ld. Something like `-sectcreate __TEXT __build_id file`. And then git_rewrite_sha.py could be modified to update it with the git SHA.